packages/shared/src/components/profile/SquadsList.tsx

```diff
@@ -0,0 +1,204 @@
+import { AuthTriggers } from '../../lib/auth';
+import { webappUrl } from '../../lib/constants';
+
+export interface SquadsListProps {
+  memberships?: Connection<SourceMember>;
+  userId: string;
+}
+
+function SquadItem({
+  membership,
+  loading,
+}: {
+  membership: SourceMember;
+  loading?: boolean;
+}): ReactElement {
+  const { user, showLogin } = useContext(AuthContext);
+  const router = useRouter();
+  const squad = membership.source;
+  const showJoin = !squad.currentMember?.role && !loading;
+  const { displayToast } = useToastNotification();
+
+  const { mutateAsync: joinSquad, isLoading } = useMutation(
+    useJoinSquad({ squad }),
+    {
+      onError: () => {
+        displayToast(labels.error.generic);
+      },
+      onSuccess: () => router.push(squad?.permalink),
+    },
+  );
+
+  const onJoin = async () => {
+    if (!user) {
+      showLogin({
+        trigger: AuthTriggers.JoinSquad,
+        options: {
+          onLoginSuccess: joinSquad,
+          onRegistrationSuccess: joinSquad,
+        },
+      });
+
+      return;
+    }
+
+    await joinSquad();
+  };
+
+  return (
+    <div
+      className="flex relative items-center p-2 first:ml-4 tablet:first:ml-0 w-40 tablet:w-auto bg-theme-float hover:bg-theme-hover rounded-2xl"
+      data-testid={squad.id}
+    >
+      <Link href={squad.permalink} prefetch={false} passHref>
+        <CardLink />
+      </Link>
+      <div className="flex overflow-hidden flex-col flex-1">
+        <div className="flex gap-2 items-center">
+          <Image
+            src={squad.image}
+            alt={squad.name}
+            className="w-8 h-8 rounded-full"
+          />
+          <div className="flex flex-col flex-1">
+            <div className="overflow-hidden font-bold whitespace-nowrap typo-caption1 text-ellipsis">
+              {squad.name}
+            </div>
+            <div className="overflow-hidden whitespace-nowrap text-theme-label-quaternary typo-caption2 text-ellipsis">
+              @{squad.handle}
+            </div>
+          </div>
+        </div>
+        <div className="flex items-center mt-1 h-6 tablet:h-auto text-theme-label-tertiary typo-caption2">
+          {membership.role !== SourceMemberRole.Member && (
+            <>
+              <SquadMemberBadge
+                role={membership.role}
+                removeMargins
+                disableResponsive
+              />
+              <span className="mx-0.5">&#x2022;</span>
+            </>
+          )}
+          <span className="overflow-hidden flex-1 whitespace-nowrap text-ellipsis">
+            {largeNumberFormat(squad.membersCount)} members
+          </span>
+          {showJoin && (
+            <Button
+              className="tablet:hidden z-1 ml-auto"
+              variant={ButtonVariant.Secondary}
+              size={ButtonSize.XSmall}
+              icon={<PlusIcon />}
+              onClick={onJoin}
+              loading={isLoading}
+              aria-label="Join Squad"
```

Comment:
Don't think you need this specifically? (but could mistake it)

Reply:
It's needed for accessibility since this button has nothing but icon